code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -588,6 +595,76 @@
 	var/area/location = get_area(src)
 	speak("Medical emergency! [crit_patient || "A patient"] is in critical condition at [location]!", radio_channel)
 
+/mob/living/simple_animal/bot/medbot/cola
+	name = "Sponsored Medibot"
+	desc = "Even medical isn't safe from corporate greed. May contain cola."
+	radio_key = /obj/item/encryptionkey/medical_sponsor  // grants cargo
+	COOLDOWN_STATIC_DECLARE(fund)
+	var/static/bot_budget = 1000  // initial value (will get refilled instantly)
+	var/static/bot_budget_refill_size = 1000
+	var/cola_cost = 250  // at start, uses entire budget in 8 heals
+	skin = "advanced"  // looks cola-y
+
+/mob/living/simple_animal/bot/medbot/cola/New()
+	START_PROCESSING(SSobj, src)
```

Comment:
Should this really process anyways? It could just randomly inject space cola, if cargo can afford it.

Reply:
it refills the local budget from cargos budget